wrapper/lib/coupler_lib.F90

```diff
@@ -668,11 +695,37 @@ subroutine get_diagnostic_2d(idx, out) bind(c)
              do i_column = 1, Atm_block%blksz(i_block) ! points per block
                 i = Atm_block%index(i_block)%ii(i_column)
                 j = Atm_block%index(i_block)%jj(i_column)
-                out(i, j) = Diag(idx)%data(i_block)%var2(i_column)
+                out(i, j) = diag_type(idx)%data(i_block)%var2(i_column)
              enddo
           enddo
        end subroutine
 
+       subroutine get_diagnostic_3d(diag_manager_controlled, idx, out) bind(c)
+          use dynamics_data_mod, only: i_start, i_end, j_start, j_end, nz
+          use atmos_model_mod, only: Atm_block
+          logical(c_int), intent(in) :: diag_manager_controlled
+          integer(c_int), intent(in) :: idx
+          real(c_double), intent(out), dimension(i_start():i_end(), j_start():j_end(), nz()) :: out
+          ! locals
+          integer :: blocks_per_MPI_domain, i, j, k, i_block, i_column, axes, n
+
+          if (diag_manager_controlled) then
+             call get_diagnostic_3d_from_diag_type(Diag_diag_manager_controlled, idx, out)
```

Comment:
[nit] Can we rename `diag_manager_controlled` to `is_diag_manager_controlled` or something that looks slightly different from `Diag_diag_manager_controlled` and makes it immediately clear that it's a boolean?

Reply:
Sure thing, makes sense!  All set in 34117a6.